solidity/deploy/33_upgrade_trove_manager.ts

```diff
@@ -0,0 +1,41 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
+  const { helpers, deployments } = hre
+
+  const { deployer } = await helpers.signers.getNamedSigners()
+
+  const { newImplementationAddress, preparedTransaction } =
+    await helpers.upgrades.prepareProxyUpgrade("TroveManager", "TroveManager", {
+      contractName: "contracts/TroveManager.sol:TroveManager",
+    })
+
+  deployments.log(
+    `prepared transaction: ${JSON.stringify(preparedTransaction)}`,
+  )
+
+  if (["mainnet", "matsnet"].includes(hre.network.name)) {
+    return
+  }
```

Comment:
We still want to verify on those networks. Additionally, not executing `deployer.sendTransaction` makes sense only on mainnet where the governance is a multisig. On matsnet, the governance (and proxy admin owner) should be set to [the deployer](https://github.com/mezo-org/musd/blob/9f818e236b137bd3424dfe71b603ecccac306376/solidity/hardhat.config.ts#L182) and we should be able to upgrade contracts there from this script. Same comment to step 32_.

Reply:
> We still want to verify on those networks.

How does that work? Maybe I'm misunderstanding what's happening here, but the `sendTransaction` is what links the implementation to the proxy, and we're not calling it yet on main, right? 

> On matsnet, the governance (and proxy admin owner) should be set to [the deployer](https://github.com/mezo-org/musd/blob/9f818e236b137bd3424dfe71b603ecccac306376/solidity/hardhat.config.ts#L182)

okay cool, i'll remove `matsnet` from the list

Reply:
e5706cd2f45ba33c71e37a60a8b3ac7fffa78d5d

Reply:
> How does that work? Maybe I'm misunderstanding what's happening here, but the sendTransaction is what links the implementation to the proxy, and we're not calling it yet on main, right?

We are not, but the verification - assuming `hre.network.tags.etherscan` is on - will verify the implementation contract. The proxy contract's ABI is different from the implementation ABI.